tests/signal_manager_test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+. ./src/signal_manager.sh --source-only
+. ./tests/utils --source-only
+
+function suite() {
+  suite_addTest "default_handler_Test"
+  suite_addTest "non_signal_Test"
+  suite_addTest "new_signal_Test"
+  suite_addTest "signal_reset_Test"
+  suite_addTest "add_two_signals_Test"
+}
+
+function default_handler_Test() {
+  local -r original_dir="$PWD"
+  local -r expected=$'\n'"Oh no! An interruption! See ya..."
+  local output
+
+  cd "$SHUNIT_TMPDIR"
+
+  {
+    signal_manager default_interrupt_handler SIGTERM SIGINT
+    sleep infinity &
+    wait $!
+  } > out.tmp &
+
+  $(echo :)
+  kill -s SIGTERM $!
```

Comment:
I'm not sure what you are trying to achieve here... are you trying to kill what you created inside `{}`? Are you sure that the PID is the same?

Reply:
Yes, one could test it with:
```
...
} > out.tmp &
local pid=$!

$(echo :)
echo "$pid should be equal to $!..."
```

That doesn't seem to be the problem.